lib/pbench/server/cache_manager.py

```diff
@@ -133,11 +116,13 @@ def __init__(self, path: Path, controller: "Controller"):
         # Record the path of the tarball file
         self.tarball_path: Path = path
 
-        # Record the unpacked INCOMING tree directory
-        self.unpacked: Optional[Path] = None
+        # Record where cached unpacked data would live
+        self.cache: Path = controller.cache / self.resource_id
 
-        # Record the RESULTS tree softlink path
-        self.results_link: Optional[Path] = None
+        # Record the base of the unpacked files for cache management, which
+        # is (self.cache / self.name) and will be None when the cache is
```

Comment:
The problem is that the indexer wants (directory)/name/<stuff...>.

That's because the tarball contents are rooted at <name> and it really works by traversing `tarfile.members` rather than a directory tree. I don't want to use the tarball *name* for my first level directory, so I chose to use the resource_id; but without significant changes, the indexer can't handle that format. I spent most of a day hacking at the indexer to figure out why it wasn't working, decided that *changing* the indexer to recognize differences between tarfile path and directory path just wasn't worth the mess, and I'd always wonder whether I missed something.

My compromise was the two-level hierarchy you see here. We point the indexer at `/srv/pbench/.cache/<resource_id>` with everything unpacked under that in `/srv/pbench/.cache/<resource_id>/<name>`, and everyone's happy.

I tried to record this in the `unpack` docstring.

Reply:
Ah yeah, that makes sense. And we shouldn't be refactoring the way the indexer works right now or until there is a real need for it.